llama/tokenizer.py

```diff
@@ -13,29 +13,32 @@
 
 class Tokenizer:
     """tokenizing and encoding/decoding text using SentencePiece."""
-    def __init__(self, model_path: str):
+    def __init__(self, model_path: Optional[str] = None):
```

Comment:
Why make this optional? How would you initialize the tokenizer without the model file?